### DISCUSSION TITLE
How to evaluate the maximum "Active Power" ever seen?

### DISCUSSION THREAD

**Discussion by ChrisDupu** [arendst/Tasmota#18020] [Q&A]
I'm using Tasmota with a small solar system with varying power input. Unfortunately it's capability and statistics data for this purpose are extreme rudimentary.

So may be, my problem can be solved somehow: I want to watch the "Active Power" value automatically to find out what the maximum value is, Tasmota ever has seen. So how can this be done? Is there a rule or something like that? And when the maximum value (for a day or in total) is retrieved - where/how can I read and reset it?

**Comment by sfromis** [comment]
My preference for such things is to have telemetry data archived in my backend, enabling me easily to go back in time to extract interesting things.

Still, it would be possible to have the rules in the Tasmota device to update a var if value larger than the old one, and reset at midnight.

**Reply by ChrisDupu** [reply]
> My preference for such things is to have telemetry data archived in my backend, enabling me easily to go back in time to extract interesting things.

This is not a solution as
a) the value can change every few seconds and therefore would cause a huge load on the network
b) it does not make sense to produce green energy and then waste loads of the produced energy for a second devices which is doing only some statistics stuff

> Still, it would be possible to have the rules in the Tasmota device to update a var if value larger than the old one, and reset at midnight.

And how can this be done?

**Reply by joba-1** [reply]
I cannot tell you exactly how but I can tell you where to look for that info: https://tasmota.github.io/docs/Rules/

From reading that it should be something like 
    Rule1 ON energy#power > %mem1% DO mem1 %value% ENDON
    Rule1 1

For the condition with variables to be usable this page also says you need to compile tasmota yourself with
    #define USE_EXPRESSION

Then you need to solve how to look at the current value of mem1. Console commands or fake sensor on web page come to mind. 

Having said that, let me comment on 
a): 
unlikely! Even a very bad wlan will give you 1 Mbit/s and with tasmota energy messages of length ~100 bytes every single second you only use 1/1000th of that. Hardly a huge load.
b) 
Even a raspberry pi zero w can already do the backend job (and collect much more statistics and present them with nice graphics and ...) for ~1-1.5W. From what you learn from these statistics you probably save more energy than that.

Have fun!

**Reply by joba-1** [reply]
It actually works :)
Tested on my **Dual**R3, hence the "[1]". You may need to remove that.
No self compile needed!?

* Collect the maximum power value 
```
rule1 on energy#power[1] > %mem1% do mem1 %value% endon
rule1 1
```
* Query the current value
```
mem1
```
* Reset the value every night at 4:00am
```
rule2 on time#minute=240 do mem1 0 endon
rule2 1
```

**Reply by sfromis** [reply]
One detail, those simple rule triggers do not need a build with `USE_EXPRESSION`, that only pertains to features within the scope of `do` and `endon`.
